client/src/app.js

```diff
@@ -203,6 +209,10 @@ export default function main({ DOM, HTTP, route, storage, scanner: scan$, search
         // use an empty object if the map fails loading for any reason
         .merge(extractErrors(HTTP.select('asset-map')).mapTo({}))
 
+  // Assets List State      
+  , assetList$ = !process.env.IS_ELEMENTS ? O.empty() : 
+      dropErrors(HTTP.select('assetlist')).map(r => ({ assets: r.body, query: r.request.query, total: r.headers['x-total-results'] || 492 }))
```

Comment:
You can use `reply()` with the second argument set to true to receive the entire response and not just the body, i.e. `reply('assetlist', true).map(r => ...)`

Reply:
Why are you keeping the request query on `assetList$` and later read it from the `asset-list` view, instead of reading it from `goAssetList`?

If that's because of my previous comment regarding putting `goAssetList` in the state - it was wrong, using `goAssetList` is fine. Please disregard what I said. :)

Reply:
changes have been made.